src/main/kotlin/org/ionproject/integration/domain/evaluations/BusinessObjects.kt

```diff
@@ -40,45 +44,48 @@ data class Evaluations(
                     jobProgramme.acronym
                 ),
                 calendarTerm,
-                buildExamList(rawEvaluationsData, jobProgramme, getCalendarYear(calendarTerm))
+                buildExamList(rawEvaluationsData, jobProgramme, getCalendarYear(calendarTerm), timeZone)
             )
         }
 
-        private fun getCalendarYear(calendarTerm: String) =
-            when (val termNumber = calendarTerm.last()) {
-                '1' -> calendarTerm.take(4)
-                '2' -> calendarTerm.substring(5, 9)
+        private fun getCalendarYear(calendarTerm: CalendarTerm) =
+            when (val termNumber = calendarTerm.term) {
+                Term.FALL -> calendarTerm.startYear.value.toString()
+                Term.SPRING -> (calendarTerm.startYear.value + 1).toString()
```

Comment:
maybe endyear should be defined as part of calendarterm instead of adding 1 to start year everywhere it's used

Reply:
But then it is necessary to validade that the endYear is the next one to the startYear. This opens to the possibility of having CalendarTerm like "2021-2050-2".

Reply:
Added `endYear`, changed the unitary tests and instantiations of the CalendarTerm.